fuel-core/src/cli/run/p2p.rs

```diff
@@ -0,0 +1,150 @@
+    /// p2p network's TCP Port
+    #[clap(long = "port", default_value = "4000")]
+    pub port: u16,
+
+    /// Max Block size
+    #[clap(long = "max_block_size", default_value = "100_000")]
+    pub max_block_size: usize,
+
+    /// Path to the location of where bootstrap nodes are listed
+    #[clap(long = "bootstrap_nodes")]
+    pub bootstrap_nodes: Option<PathBuf>,
+
+    /// Allow nodes to be discoverable on the local network
+    #[clap(long = "enable_mdns")]
+    pub enable_mdns: bool,
+
+    /// Maximum amount of allowed connected peers
+    #[clap(long = "max_peers_connected", default_value = "50")]
+    pub max_peers_connected: usize,
+
+    /// Enable random walk for p2p node discovery
+    #[clap(long = "enable_random_walk")]
+    pub enable_random_walk: bool,
+
+    /// Choose to include private IPv4/IPv6 addresses as discoverable
+    /// xcept for the ones stored in `bootstrap_nodes`
+    #[clap(long = "allow_private_addresses")]
+    pub allow_private_addresses: bool,
+
+    /// Choose how long will connection keep alive if idle
+    #[clap(long = "allow_private_addresses", default_value = "120")]
+    pub connection_idle_timeout: u64,
+
+    /// Choose how often to recieve PeerInfo from other nodes
+    #[clap(long = "info_interval", default_value = "3")]
+    pub info_interval: u64,
+
+    /// Choose the interval at which identification requests are sent to
+    /// the remote on established connections after the first request
+    #[clap(long = "identify_interval", default_value = "5")]
+    pub identify_interval: u64,
+
+    /// Choose which topics to subscribe to via gossipsub protocol
+    #[clap(long = "topics", default_value = "new_tx, new_block, consensus_vote")]
+    pub topics: Vec<String>,
+
+    /// Choose max mesh size for gossipsub protocol
+    #[clap(long = "max_mesh_size")]
+    pub max_mesh_size: usize,
+
+    /// Choose min mesh size for gossipsub protocol
+    #[clap(long = "min_mesh_size")]
+    pub min_mesh_size: usize,
+
+    /// Choose ideal mesh size for gossipsub protocol
+    #[clap(long = "ideal_mesh_size")]
+    pub ideal_mesh_size: usize,
+
+    /// Choose timeout for sent requests in RequestResponse protocol
+    #[clap(long = "request_timeout")]
+    pub request_timeout: Option<u64>,
+
+    /// Choose how long RequestResponse protocol connections will live if idle
+    #[clap(long = "connection_keep_alive")]
+    pub connection_keep_alive: Option<u64>,
+}
+
+impl From<P2pArgs> for anyhow::Result<P2PConfig> {
+    fn from(args: P2pArgs) -> Self {
+        let local_keypair = {
+            match args.keypair {
+                Some(path) => {
+                    let mut bytes = std::fs::read(path)?;
+                    Keypair::secp256k1_from_der(&mut bytes)?
+                }
+                _ => Keypair::generate_secp256k1(),
+            }
+        };
+
+        let bootstrap_nodes = {
+            match args.bootstrap_nodes {
+                Some(path) => std::fs::read_to_string(path)?
```

Comment:
I'd prefer taking the bootnodes via comma separated cli args instead of loading from a file.

1. There is no snapshot test of the file format or automated tests that exercise this file parsing works as expected
2. For small network configurations it's inconvenient to require a separate dedicated config file just to pass a handful of bootstrap URLs
3. We have other tasks related to supporting a config file-based approach (#458), and I think long lists of bootnode URLs should be included in the main YAML/TOML for the whole service instead of an independent config file format.